packages/rich-text/src/component/use-input-and-selection.js

```diff
@@ -254,32 +273,23 @@ export function useInputAndSelection( props ) {
 			// at this point, but this focus event is still too early to calculate
 			// the selection.
 			rafId = defaultView.requestAnimationFrame( handleSelectionChange );
-
-			ownerDocument.addEventListener(
-				'selectionchange',
-				handleSelectionChange
-			);
-		}
-
-		function onBlur() {
-			ownerDocument.removeEventListener(
-				'selectionchange',
-				handleSelectionChange
-			);
 		}
 
 		element.addEventListener( 'input', onInput );
 		element.addEventListener( 'compositionstart', onCompositionStart );
 		element.addEventListener( 'compositionend', onCompositionEnd );
 		element.addEventListener( 'focus', onFocus );
-		element.addEventListener( 'blur', onBlur );
```

Comment:
What were blurs aiming to achieve before?

Reply:
Removing `selectionchange` event listeners. But now we care about selection outside of rich text as well. More precisely selection that starts in rich text and ends in another. This selection state needs to be communicated to the block editor store so we can perform the keyboard actions on the selection.